src/config.ts

```diff
@@ -328,3 +332,24 @@ export async function urlFromLocalPath(
   }
   return urlPath;
 }
+
+export async function writeBackSchemaIfNeeded(
+    rawConfigObj: Partial<ConfigFile>, configFile: string) {
+  // Add the $schema field to the original config file if it's absent.
+  // We only want to do this if the file validated though, so we don't mutate
+  // a file that's not actually a tachometer config file.
+  if (!('$schema' in rawConfigObj) ||
+      (rawConfigObj.$schema &&
+       rawConfigObj.$schema.startsWith('https://unpkg.com/tachometer'))) {
+    // Extra IDE features can be activated if the config file has a schema.
+    const withSchema = {
+      '$schema': ``,  // write it empty, to get the key ordering correct
+      ...rawConfigObj,
+    };
+    // Then write the value, ensuring that we overwrite rawConfigObj.$schema
+    withSchema.$schema = `https://unpkg.com/tachometer@${
+        tachometerVersion}/lib/config.schema.json`;
```

Comment:
It's kind of unfortunate that this is going to cause frequent churn for user config files, and also inconsistently since it will only update the schema after you execute in tachometer.

How about we just check the generated schema into master, and use a `raw.githubusercontent.com` URL instead? Users will always get the latest one, but that seems like a reasonable tradeoff?

Reply:
Maybe another option is to use a local file path? This seems to work for me in VSCode:

```
  "$schema": "./node_modules/tachometer/lib/config.schema.json",
```

Reply:
Not blocking if you still think it's the best way. I can just imagine getting a little annoyed by having my config files frequently changing unexpectedly. Note it also means if my editor's JSOn formatting doesn't exactly match `JSON.stringify(indent=2)`, it's going to sometimes churn the whole file.

Another totally different idea, it seems that VSCode supports http://schemastore.org/json/, which maintains file pattern to schema mappings. So what if we had a `foo.tachometer.json` convention and just add an entry there?

Reply:
I like the githubusercontent model. No objection to schemastore.org as well though.